File: Algorithms/Python/Minimum-Depth-of-Binary-Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution(object):

    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def getDepth(node, depth):
            depth += 1
            if not node.left and not node.right:
                return depth
            elif node.left and not node.right:
                return getDepth(node.left, depth)
            elif not node.left and node.right:
                return getDepth(node.right, depth)
            else:
                return min(getDepth(node.left, depth), getDepth(node.right, depth))
        if not root:
            return 0
        return getDepth(root, 0)

File: Algorithms/Java/Minimum Depth of Binary Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int minDepth(TreeNode root) {
		if (root == null)
			return 0;
		return find(root, 1);
	}

	public int find(TreeNode tree, int n) {
		if (tree.left != null && tree.right != null)
			return Math.min(find(tree.left, n + 1), find(tree.right, n + 1));
		else if (tree.left != null)
			return find(tree.left, n + 1);
		else if (tree.right != null)
			return find(tree.right, n + 1);
		else
			return n;
	}

}